src/main/webapp/WEB-INF/view/profile.jsp

```diff
@@ -24,36 +29,55 @@ UserStore userStore = UserStore.getInstance();
       String currentUser = (String) request.getSession().getAttribute("user");
       String profileUser = (String) request.getAttribute("profilePage");
       User thisUser = userStore.getUser(profileUser);
+
       if (profileUser.equals("")) { %>
         <h1>User does not exist.</h1>
 
-      <% } else if (currentUser != null && currentUser.equals(profileUser)) { %>
-        <h2 style="color:blue">Welcome to your page!</h2>
+      <% } else {
+        if (currentUser != null && currentUser.equals(profileUser)) { %>
+            <h1 style="color:dodgerblue">Welcome to your page!</h1>
+        <% } else { %>
+            <h1 style="color:dodgerblue">Welcome to <%= profileUser %>'s Page!</h1>
+        <% } %>
         <% /** Gets the bio of this user to display on their profile page */ %>
-        <% String profilePageBio = thisUser.getBio(); %>
-        <h3 style="color:purple">About Me</h3>
-        <a> <%= profilePageBio %> </a>
-        <br/>
-        <br/>
-
-        <% /** Gives current user a form that allows them to edit their own bio */ %>
-        <a> Edit your bio here! (only you can see this) </a>
-        <form action="/users/<%= currentUser %>" method="POST">
-            <input type="text" name="bio" value="<%= thisUser.getBio() %>" >
+            <% String profilePageBio = thisUser.getBio(); %>
+            <h2 style="color:skyblue">About Me</h2>
+            <a> <%= profilePageBio %> </a>
             <br/>
-            <button type="submit">Submit</button>
-        </form>
+            <br/>
+        <% if (currentUser != null && currentUser.equals(profileUser)) { %>
+            <% /** Gives current user a form that allows them to edit their own bio */ %>
+                <a> Edit your bio here! (only you can see this) </a>
+                <form action="/users/<%= currentUser %>" method="POST">
+                    <input type="text" name="bio" value="<%= thisUser.getBio() %>" >
+                    <br/>
+                <button type="submit">Submit</button>
+                </form>
+         <% } %>
+         <h2 style="color:skyblue"> <%= profileUser %>'s Sent Messages </h2>
+         <% List<Message> userMessages = (List) request.getAttribute("messages");
```

Comment:
I've seen this in other jsp files, but what does the (List) part do? Does it cast what you're returning from request.getAttribute() into a list?

Reply:
Yeah it casts the return value from the request! I think it's because the getAttribute method only returns a generic Object and since I know it will be a List, it's okay to cast it here.

Reply:
Okay that's what I was thinking, I have to tinker around with the attribute stuff more to really understand it.